src/ui/components/tags/TagEditor.vue

```diff
@@ -92,57 +88,67 @@ export default {
         }
     },
     watch: {
-        annotation: {
-            handler() {
-                this.tagsChanged(this.annotation.tags);
-            },
-            deep: true
-        },
-        annotationQuery: {
+        annotations: {
             handler() {
-                this.unloadAnnotation();
-                this.loadAnnotation();
+                this.annotationsChanged();
             },
             deep: true
         }
     },
     mounted() {
-        this.loadAnnotation();
+        this.annotationDeletionListener = this.annotationDeletionListener.bind(this);
```

Comment:
I believe Vue component methods [are automatically bound to the viewmodel](https://vuejs.org/api/options-state.html#methods).